sorc/ncep_post.fd/SURFCE.f

```diff
@@ -1400,7 +1400,7 @@ SUBROUTINE SURFCE
             GRID2=SPVAL
             DO J=JSTA,JEND
             DO I=1,IM
-            if(TSHLTR(I,J)/=spval) then
+            if(TSHLTR(I,J)/=spval.and.PSHLTR(I,J)/=spval.and.QSHLTR(I,J)/=spval) then
```

Comment:
Change SHLTR to EGRID1.

Reply:
grid1 is initialized with spval. so if (egrid1==spval) grid1=egrid1 still valid.